pkg/cmd/pulumi/stack_init.go

```diff
@@ -172,5 +181,55 @@ func newStackInitCmd() *cobra.Command {
 		&stackToCopy, "copy-config-from", "", "The name of the stack to copy existing config from")
 	cmd.PersistentFlags().BoolVar(
 		&noSelect, "no-select", false, "Do not select the stack")
+	cmd.PersistentFlags().StringArrayVar(&teams, "teams", nil, "A list of team "+
+		"names that should have permission to read and update this stack,"+
+		" once created")
 	return cmd
 }
+
+// This function constructs a createStackOptions object if
+// valid, otherwise returning nil. Most backends expect nil
+// options, and error if options are non-nil.
+func validateCreateStackOpts(stackName string, b backend.Backend, teams []string) (*backend.CreateStackOptions, error) {
+	// • If the user provided teams but the backend doesn't support them,
+	//   return an error.
+	if len(teams) > 0 && !b.SupportsTeams() {
+		return nil, newTeamsUnsupportedError(stackName, b.Name())
+	}
+	// • Otherwise, validate the teams and pass them along.
+	//   Remove any strings from the list that are empty or just whitespace.
+	validatedTeams := teams[:0] // reuse storage.
+	for _, team := range teams {
+		teamStr := strings.TrimSpace(team)
+		if len(teamStr) > 0 {
+			validatedTeams = append(validatedTeams, teamStr)
+		}
+	}
+
+	// • We can return stack options that contain the provided teams.
+	//   since this will be zerod for non-Service backends.
+	return &backend.CreateStackOptions{
+		Teams: validatedTeams,
+	}, nil
+}
+
+// TeamsUnsupportedError is the error returned when the --teams
+// flag is provided on a backend that doesn't support teams.
+type teamsUnsupportedError struct {
```

Comment:
Not sure we really need an error struct for this? Not sure in general what our leaning is vis-a-vis making new error types or just using fmt.Errorf actually...

Reply:
Generally, we don't need to bother with new error types for short-enough messages,
where the caller doesn't need to match on that exact object and extract information
([relevant ref](https://abhinav.github.io/presentation-go-patterns-2022/#/structured-errors)).
However, if the error message is complex/has a bunch of conditionals to determine what gets put in it,
an error object is worth.

In this case, we don't need it, but I don't consider it a blocker by any means.

Reply:
Let's discuss in ideation.